### DISCUSSION TITLE
Cross DC cluster configuration

### DISCUSSION THREAD

**Discussion by pavlenkoms** [ClickHouse/ClickHouse#59045] [Q&A]
Hi
The team mentioned that CH has a feature of cross DC master-master replication (even for DCs in multiple countries), and I haven't found any manual or schema or picture or advice how to make it working.
Does it have the same approach as an usual single DC cluster with 3 keeper nodes but spread over different DCs without any extra tuning?
At the same time was mentioned [here](https://clickhouse.com/docs/en/faq/operations/multi-region-replication) that it's not always the case because of the requirement for <100ms latency.

**Comment by UnamedRus** [comment]
>  with 3 keeper nodes but spread over different DCs without any extra tuning?

It's more like 3 keeper quorum nodes in single DC, so inserts (or at least inserts with high frequency) goes to that particular DC.

**Comment by masha00987** [comment]
For cross-DC master-master replication in CH, you generally need to set up a cluster with the same approach as a single DC, but you must account for latency requirements. Ensure all DCs meet the <100ms latency requirement and consult the ClickHouse documentation or support for specific configuration details.
https://www.vancorr.com/
